client/data/plugins/use-update-schedules-capabilities-query.ts

```diff
@@ -11,10 +11,10 @@ export type ScheduleUpdatesCapabilities = {
 export const useUpdateScheduleCapabilitiesQuery = (
 	siteSlug: SiteSlug,
 	isEligibleForFeature: boolean
-): UseQueryResult< ScheduleUpdatesCapabilities > => {
+) => {
 	return useQuery( {
 		queryKey: [ 'schedule-updates-capabilities', siteSlug ],
-		queryFn: () =>
+		queryFn: (): Promise< ScheduleUpdatesCapabilities > =>
 			wpcomRequest( {
 				path: `/sites/${ siteSlug }/update-schedules/capabilities`,
 				apiNamespace: 'wpcom/v2',
```

Comment:
This represents a typical type fix. Rather than have the hook return a result, it is better to confirm the queryFn is returning the correct type, and then all the types flow through the code fine.